docs/source/roadmap.md

```diff
@@ -0,0 +1,287 @@
+
+### A Modular Library of Pipeline Components
+_starfish_ enables the construction of arbitrary image-based transcriptomics pipelines by implementing the union of algorithms used by SpaceTx labs while combining similar approaches into a simplified subset. These algorithms can be linearly combined to process individual fields of view, requiring no more than 16GB ram for any individual pipeline component. The set of algorithms must include:
+
+#### Image Adjustment
+_starfish_ defines pipeline components that enable simple adjustment of image data, such as slicing (finding focal plane, cropping edge effects) and projection over arbitrary axes.
+
+#### Filtering
+_starfish_ defines the set of filters used in SpaceTx labs, including approaches to remove background, enhance spots, deconvolve the specified point spread function of the imaging system, and normalize image intensities.
+
+#### Spot Detection
+_starfish_ implements spot detectors via a variety of spot detection algorithms (eg., Gaussian/Laplacian of Gaussian/Hessian of Gaussian blob detection, local max peak finding, etc.) These detectors can function independently on individual channels and rounds or be used to measure intensity values of spots (as detected in a single auxiliary image) across all other primary images across channels and rounds. _starfish_ also supports pixel based decoding algorithms (e.g., algorithms that detect spots after pixels are decoded to targets) within the same API.
+
+#### Decoding
+_starfish_ enables decoding of spots or pixels detected across rounds and channels into targets. These decoding results can be filtered post-hoc, using spot intensity, spot size, and spot distance-to-target metrics. For the latter metric, the user may supply their desired distance metric.
+
+#### Segmentation
+This is an area of active research. However, four SpaceTx groups use seeded watershed segmentation, which has a concrete implementation. _starfish_ implements this method and will include a vignette on how to tune parameters so it functions across assays. In addition, we will elicit submissions from Kenneth Harris and Peter Kharchenko for point-cloud based segmentation, and encourage the community to work on this problem by defining a segmentation format and API to enable plugin-based interaction with community-developed algorithms.
+
+#### Image Registration
+_starfish_ implements the ability to apply a pre-computed full affine registration including translation, rotation, and scale to enable alignment across z-layers, rounds, and channels. Translation and rotation can also provide a solution for some types of chromatic aberration.
```

Comment:
wait it does?

or this is something we plan to do?  if the latter, should be future tense.